tlatools/org.lamport.tlatools/src/tla2sany/utilities/Vector.java

```diff
@@ -1,143 +1,100 @@
 // Copyright (c) 2003 Compaq Corporation.  All rights reserved.
 package tla2sany.utilities;
+
+import java.util.ArrayList;
 import java.util.Enumeration;
+import java.util.Iterator;
+
+/**
+ * Legacy wrapper around {@link ArrayList} to preserve historical semantics:
+ * <ul>
+ *   <li>{@link #contains(Object)} uses reference equality (==), not {@link Object#equals(Object)}.</li>
+ *   <li>{@link #insertElementAt(Object, int)} rejects {@code index == size()} (cannot append via insert).</li>
+ *   <li>{@link #elements()} returns a snapshot Enumeration taken at call time.</li>
+ * </ul>
+ * These differences make a drop-in replacement with {@link ArrayList} unsafe until all call sites
+ * are audited for identity-based membership checks.
+ */
+public class Vector<E> extends ArrayList<E> {
+  private static final long serialVersionUID = 5334890925756142170L;
```

Comment:
Including `serialVersionUID` is fine but unnecessary. There's a good overview [here](https://github.com/projectlombok/lombok/wiki/WHY-NOT:-serialVersionUID):

> (almost) all tools are wrong: They ship with code quality plugins that remark on serializable classes without a serialVersionUID field as being wrong. This is incorrect; the vast majority of your serializable classes **should not have a serialVersionUID field**, because this gets you the desired default behaviour of no binary compatibility [when you change the class].

Reply:
Thank you for this very interesting feedback